graph/db/benchmark_test.go

```diff
@@ -0,0 +1,703 @@
+// copying all nodes and channels from the source to the destination.
+func syncGraph(t *testing.T, src, dest *ChannelGraph) {
+	ctx := context.Background()
+
+	var (
+		s = rate.Sometimes{
+			Interval: 10 * time.Second,
+		}
+		t0 = time.Now()
+
+		chunk = 0
+		total = 0
+		mu    sync.Mutex
+	)
+
+	reportNodeStats := func() {
+		elapsed := time.Since(t0).Seconds()
+		ratePerSec := float64(chunk) / elapsed
+		t.Logf("Synced %d nodes (last chunk: %d) "+
+			"(%.2f nodes/second)",
+			total, chunk, ratePerSec)
+
+		t0 = time.Now()
+	}
+
+	var wgNodes sync.WaitGroup
+	err := src.ForEachNode(ctx, func(tx NodeRTx) error {
+		wgNodes.Add(1)
+		go func() {
+			defer wgNodes.Done()
+
+			err := dest.AddLightningNode(
+				ctx, tx.Node(), batch.LazyAdd(),
+			)
+			require.NoError(t, err)
+
+			mu.Lock()
+			total++
+			chunk++
+			s.Do(func() {
+				reportNodeStats()
+				chunk = 0
+			})
+			mu.Unlock()
+		}()
+
+		return nil
+	}, func() {})
+	require.NoError(t, err)
+
+	wgNodes.Wait()
+	reportNodeStats()
+	t.Logf("Done syncing %d nodes", total)
+
+	total = 0
+	chunk = 0
+	t0 = time.Now()
+
+	reportChanStats := func() {
+		elapsed := time.Since(t0).Seconds()
+		ratePerSec := float64(chunk) / elapsed
+		t.Logf("Synced %d channels (and its "+
+			"policies) (last chunk: %d) "+
+			"(%.2f channels/second)",
+			total, chunk, ratePerSec)
+
+		t0 = time.Now()
+	}
+
+	var wgChans sync.WaitGroup
+	err = src.ForEachChannel(ctx, func(info *models.ChannelEdgeInfo,
+		policy1, policy2 *models.ChannelEdgePolicy) error {
+
+		// Add each channel & policy. We do this in a goroutine to
+		// take advantage of batch processing.
+		wgChans.Add(1)
+		go func() {
+			defer wgChans.Done()
+
+			err := dest.AddChannelEdge(
+				ctx, info, batch.LazyAdd(),
+			)
+			if !errors.Is(err, ErrEdgeAlreadyExist) {
+				require.NoError(t, err)
+			}
+
+			if policy1 != nil {
+				err = dest.UpdateEdgePolicy(
+					ctx, policy1, batch.LazyAdd(),
+				)
+				require.NoError(t, err)
+			}
+
+			if policy2 != nil {
+				err = dest.UpdateEdgePolicy(
+					ctx, policy2, batch.LazyAdd(),
+				)
+				require.NoError(t, err)
+			}
+
+			mu.Lock()
+			total++
+			chunk++
+			s.Do(func() {
+				reportChanStats()
+				chunk = 0
+			})
+			mu.Unlock()
+		}()
+
+		return nil
+	}, func() {})
+	require.NoError(t, err)
+
+	wgChans.Wait()
+	reportChanStats()
+
+	t.Logf("Done syncing %d channels", total)
+}
+
+// BenchmarkCacheLoading benchmarks how long it takes to load the in-memory
+// graph cache from a populated database.
+//
+// NOTE: this is to be run against a local graph database. It can be run
+// either against a kvdb-bbolt channel.db file, or a kvdb-sqlite channel.sqlite
+// file or a postgres connection containing the channel graph in kvdb format and
+// finally, it can be run against a native SQL sqlite or postgres database.
+func BenchmarkCacheLoading(b *testing.B) {
+	ctx := context.Background()
+
+	tests := []dbConnection{
```

Comment:
So I guess the idea is that we would run `TestPopulateDBs` first from a single KVDB source to create all other sources?
Or provide a manual test source. Perhaps we can add that to the `NOTE` above?

I think it would be good if at least all team members could run these tests on their own hardware with their own mainnet node data to get some good numbers. So making it as easy as possible to run them will probably be helpful.